### ISSUE TITLE
Deprecate assertions that operate on (non-public) attributes

### ISSUE THREAD

**Issue by sebastianbergmann** [sebastianbergmann/phpunit#3338]
The following assertions should be deprecated:

* `assertAttributeContains()`
* `assertAttributeNotContains()`
* `assertAttributeContainsOnly()`
* `assertAttributeNotContainsOnly()`
* `assertAttributeCount()`
* `assertAttributeNotCount()`
* `assertAttributeEquals()`
* `assertAttributeNotEquals()`
* `assertAttributeEmpty()`
* `assertAttributeNotEmpty()`
* `assertAttributeGreaterThan()`
* `assertAttributeGreaterThanOrEqual()`
* `assertAttributeLessThan()`
* `assertAttributeLessThanOrEqual()`
* `assertAttributeSame()`
* `assertAttributeNotSame()`
* `assertAttributeInstanceOf()`
* `assertAttributeNotInstanceOf()`
* `assertAttributeInternalType()`
* `assertAttributeNotInternalType()`

**Comment by TomasVotruba** [comment]
What should be used inste?

**Comment by gmponos** [comment]
https://github.com/sebastianbergmann/phpunit/issues/3339

**Comment by TomasVotruba** [comment]
@gmoponos Thanks!

**Comment by geoidesic** [comment]
How do I figure out what replaces these?

**Comment by sebastianbergmann** [comment]
@geoidesic There is no replacement. The correct course of action is to refactor your code to not require dirty tricks like this.

**Comment by aogg** [comment]
> @geoidesic There is no replacement. The correct course of action is to refactor your code to not require dirty tricks like this.

why not

**Comment by AlexMalikov94** [comment]
@sebastianbergmann What alternative using ```readAttribute()```. If it deprecated almost for a secure reason what method I just need for use?

**Comment by sebastianbergmann** [comment]
@AlexMalikov94 There is no replacement. The correct course of action is to refactor your code to not require dirty tricks like this.

**Comment by AlexMalikov94** [comment]
@sebastianbergmann Yes you are right is dirty tricks but in the past developers used it just I think you have an alternative I see it is not. I just use another function more flexibly. I will contribute your code in the future.
